garden_ai/app/prototype.py

```diff
@@ -246,64 +259,89 @@ def _funcx_invoke_pipeline(*args, **kwargs):
     dill.load_session("session.pkl")
 
     for obj in globals().values():
-        if getattr(obj, "__name__", None) == "garden_target" and getattr(
-            obj, "_check", None
+        if (
+            getattr(obj, "__name__", None) == "garden_target"
+            and (meta := getattr(obj, "_pipeline_meta", None)) is not None
+            and meta["short_name"] == short_name
         ):
             target_func = obj
             break
     else:
-        raise ValueError("No function marked for invocation.")
+        raise ValueError(
+            "No function matching the requested name was marked for invocation."
+        )
 
     return target_func(*args, **kwargs)
 
 
+def _curry(func, arg):
+    def wrapper(*args, **kwargs):
+        return func(arg, *args, **kwargs)
+
+    return wrapper
+
+
 @prototype_app.command(no_args_is_help=True)
 def register(
     image: str = typer.Argument(
         ...,
         help=("The name of the image to be pushed and registered with Globus compute."),
-    ),
-    record: Path = typer.Argument(
-        ...,
-        dir_okay=False,
-        file_okay=True,
-        resolve_path=True,
-        help=("Path to a json record describing the metadata for your pipeline."),
-    ),
+    )
 ):
-    if not record.exists() or not record.is_file():
-        console.log(f"{record} does not appear to be a valid path.")
-        raise typer.Exit(code=1)
-
-    with open(record, "r") as f:
-        meta = json.load(f)
-
-    client = GardenClient()
-
-    # add container to docker registry (when updating the container, the name MUST be changed or the cache lookup will find old version)
+    # add container to docker registry
+    # when updating the container, the name MUST be changed or the cache lookup will find old version
     subprocess.run(["docker", "tag", f"{IMAGE_NAME}-planted", image])
     subprocess.run(["docker", "push", image])
     subprocess.run(["docker", "rmi", image])
 
-    import __main__
+    client = GardenClient()
 
-    # perform container and function registration
     container_id = client.compute_client.register_container(
         f"docker.io/{image}", "docker"
     )
-    meta["container_id"] = container_id
+    container_uuid = UUID(container_id)
+    # print(f"Your container has been registered with UUID: {container_id}")
+
+    pipeline_metas = []
+    total_meta = _extract(image)
+    for key, meta in total_meta.items():
+        if "." in key:  # ignore connectors metadata
```

Comment:
How does the period indicate that it's connectors?

Reply:
The naming scheme for the keys is laid out in `_extract_metadata_from_planted_container`, basically its the `pipeline.short_name` for the pipeline metadata (which I don't believe can contain periods since it must be an identifier) and `f"{pipeline.short_name}.connectors"` for the pipeline's connectors.

Reply:
Gotcha, thanks!